apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/SpannerCommandTests.cs

```diff
@@ -85,5 +92,114 @@ public void CloneUpdateWithParameters()
         }
 
+        [Fact]
+        public void CommandHasConnectionQueryOptions()
+        {
+            const string optimizerVersion = "1";
+            Mock<SpannerClient> spannerClientMock = SetupExecuteStreamingSql(optimizerVersion);
+
+            SpannerConnection connection = BuildSpannerConnection(spannerClientMock);
+            var queryOptions = new QueryOptions().WithOptimizerVersion(optimizerVersion);
+            connection.QueryOptions = queryOptions;
+
+            var command = connection.CreateSelectCommand("SELECT * FROM FOO");
+            using (var reader = command.ExecuteReader())
+            {
+                // Do nothing.
+            }
+        }
+
+        [Fact]
+        public void CommandHasOptimizerVersionFromEnvironment()
+        {
+            // Save existing value of environment variable.
+            const string optimizerVersionVariable = "SPANNER_OPTIMIZER_VERSION";
+            string savedOptimizerVersion = Environment.GetEnvironmentVariable(optimizerVersionVariable);
+            const string envOptimizerVersion = "2";
+            Environment.SetEnvironmentVariable(optimizerVersionVariable, envOptimizerVersion);
+
+            // Optimizer version set through environment variable has higher
+            // precedence than version set through connection.
+            Mock<SpannerClient> spannerClientMock = SetupExecuteStreamingSql(envOptimizerVersion);
+
+            const string optimizerVersion = "1";
+            SpannerConnection connection = BuildSpannerConnection(spannerClientMock);
+            var queryOptions = new QueryOptions().WithOptimizerVersion(optimizerVersion);
+            connection.QueryOptions = queryOptions;
+
+            var command = connection.CreateSelectCommand("SELECT * FROM FOO");
+            using (var reader = command.ExecuteReader())
+            {
+                // Do nothing.
+            }
+
+            // Set the environment back.
+            Environment.SetEnvironmentVariable(optimizerVersionVariable, savedOptimizerVersion);
+        }
+
+        [Fact]
+        public void CommandHasOptimizerVersionSetOnCommand()
+        {
+            // Save existing value of environment variable.
+            const string optimizerVersionVariable = "SPANNER_OPTIMIZER_VERSION";
+            string savedOptimizerVersion = Environment.GetEnvironmentVariable(optimizerVersionVariable);
+            const string envOptimizerVersion = "2";
+            Environment.SetEnvironmentVariable(optimizerVersionVariable, envOptimizerVersion);
+
+            var cmdOptimizerVersion = "3";
+            // Optimizer version set at a command level has higher precedence
+            // than version set through the connection or the environment
+            // variable.
+            Mock<SpannerClient> spannerClientMock = SetupExecuteStreamingSql(cmdOptimizerVersion);
+
+            const string optimizerVersion = "1";
```

Comment:
Nit: rename to `connOptimizerVersion` to clarify? The other two variables are pleasantly clear :)